addons/mail/controllers/discuss/poll.py

```diff
@@ -0,0 +1,72 @@
+from markupsafe import Markup
+from werkzeug.exceptions import NotFound
+
+from odoo.addons.mail.tools.discuss import add_guest_to_context, Store
+from odoo.fields import Command
+from odoo.http import Controller, route
+
+
+class PollController(Controller):
+    @route("/discuss/poll/create", type="jsonrpc", auth="user", methods=["POST"])
+    def poll_create(self, answers, channel_id, duration, question, allow_multiple_answers=False):
+        channel = self.env["discuss.channel"].search_fetch([("id", "=", channel_id)])
+        if not channel:
+            raise NotFound()
+        notification = Markup(
+            "<div class='o_mail_notification o-discuss-Poll-started'>%s</div>"
+        ) % self.env._("created a poll: %s.", Markup("<strong>%s</strong>") % question)
+        message = channel.message_post(
+            body=notification, message_type="notification", subtype_xmlid="mail.mt_comment"
+        )
+        poll = self.env["discuss.poll"].create(
+            {
+                "allow_multiple_answers": allow_multiple_answers,
+                "answer_ids": [Command.create({"text": answer}) for answer in answers],
+                "duration": duration,
+                "question": question,
+                "start_message_id": message.id,
+            }
+        )
+        Store(bus_channel=poll).add(poll).bus_send()
+
+    @route("/discuss/poll/vote", type="jsonrpc", auth="public", methods=["POST"])
+    @add_guest_to_context
+    def poll_vote(self, poll_id, answer_ids):
+        answers = self.env["discuss.poll.answer"].search_fetch(
+            [("poll_id", "=", poll_id), ("id", "in", answer_ids)]
+        )
+        if len(answers) != len(answer_ids):
+            raise NotFound()
```

Comment:
Isn't this the same? I don't get the check on `answer_ids`, what are you trying to do?
```suggestion
        poll = self.env["discuss.poll"].search_fetch(["id", "=", poll_id])
        if not poll or len(poll.answer_ids) != len(answer_ids):
            raise NotFound()
```